### PR TITLE
Feat/dao profile

### PR DESCRIPTION
**Pull request by angeljgomezc**

### CODE REVIEW

File: src/components/manage_dao/edit_form.tsx [hunk 1]
```diff
@@ -1,302 +1,659 @@
...
+
+    if (changedPartners || changedPartnersImage) {
+      let oldPartners = [...partners!]
+      let partnerImageHash = '';
+      await Promise.all(oldPartners.map(async (partner, index) => {
+        partnerImageHash = (await uploadToIPFS(partner?.image!)).hash
+        oldPartners[index].image = partnerImageHash;
+        updateFormData({ partners: oldPartners });
+      }))
+    }
```
Comment (angeljgomezc) [inline src/components/manage_dao/edit_form.tsx:211]: hey @adiwajshing - looking to add partners and an encoded about to the smart contract request but struggling to send everything at the same time. It seems like the `uploadToIPFS` function takes a while and by the time the `handleSubmit` function initiates the update request, the IPFS hash is still not stored on the `editedFormData`.

I've been stuck on this for a bit, would love if you could give me a hand here - what do you recommend? Should I handle the IPFS upload on the `useWorkspaceUpdate` hook instead?

File: src/components/manage_dao/edit_form.tsx [hunk 2]
```diff
@@ -1,172 +1,328 @@
...
+		)
+	)
+
+	const [partnersRequired, setPartnersRequired] = React.useState(false)
+	const [partners, setPartners] = React.useState<PartnersProps[] | undefined>([])
+	const [changedPartners, setChangedPartners] = React.useState(false)
+	const [changedAbout, setChangedAbout] = React.useState(false)
+	const [changedPartnersImage, setChangedPartnersImage] = React.useState(false)
+
+	React.useEffect(() => {
```
Comment (adiwajshing) [inline src/components/manage_dao/edit_form.tsx:77]: redundant log

File: src/components/manage_dao/edit_form.tsx [hunk 3]
```diff
@@ -1,172 +1,328 @@
...
+		if(changedAbout) {
+			const newAboutString = await JSON.stringify(
+				convertToRaw(newAbout.getCurrentContent())
+			)
+
+			const newAboutHash = (await uploadToIPFS(newAboutString)).hash
+			updateFormData({ about: newAboutHash })
+		}
+
+		if(changedPartners || changedPartnersImage) {
```
Comment (adiwajshing) [inline src/components/manage_dao/edit_form.tsx:203]: should put this at the end after everything has been validated, will return in unnecessary requests

File: src/components/manage_dao/edit_form.tsx [hunk 4]
```diff
@@ -1,172 +1,328 @@
...
-				image: imageFile!,
-				coverImage: coverImageFile!,
-				twitterHandle,
-				discordHandle,
-				telegramChannel,
+		if(!editedFormData?.about?.startsWith('Qm')) {
+			return updateEditError('about', 'Please enter about')
+		}
+
+		console.log(editedFormData)
```
Comment (adiwajshing) [inline src/components/manage_dao/edit_form.tsx:225]: also redundant log

File: pages/embed.tsx [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+
+	const [chainID, setChainId] = useState<SupportedChainId>()
+	const [daoID, setDaoId] = useState<string>()
+	const [workspaceData, setWorkspaceData] = useState<DAOWorkspace>()
+	const [grantsApplicants, setGrantsApplicants] = useState<any>([])
+	const [grantsDisbursed, setGrantsDisbursed] = useState<any>([])
+	const [grantRecipients, setGrantRecipients] = useState<any>([])
+
+	useEffect(() => {
+		if(router && router.query) {
```
Comment (adiwajshing) [inline pages/embed.tsx:42]: can just write `router?.query`

File: src/components/explore_grants/about_grant/grantDetails.tsx [hunk 6]
```diff
@@ -27,11 +27,13 @@ function GrantDetails({
...
 		}
 
-		if(grantDetails.startsWith('Qm') && grantDetails.length < 64) {
+		if(grantDetails.length) {
 			getDecodedDetails(grantDetails)
 		} else {
 			setDecodedDetails(grantDetails)
 		}
+
+		console.log(grantDetails)
```
Comment (adiwajshing) [inline src/components/explore_grants/about_grant/grantDetails.tsx:36]: redundant console.log

File: src/components/profile/dao_data.tsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+	useEffect(() => {
+		console.log(grants)
+		console.log(disbursed)
+
+		if(grants && grants.length > 0) {
+			grants.grants.forEach((grant: any) => setGrantoCheck((array: any) => [...array, grant.createdAtS])
+			)
+		}
+
+		console.log(grantToCheck)
```
Comment (adiwajshing) [inline src/components/profile/dao_data.tsx:28]: redundant

File: src/components/profile/grantCard/index.tsx [hunk 8]
```diff
@@ -50,29 +49,36 @@ function BrowseGrantCard({
...
 	onTitleClick,
 }: BrowseGrantCardProps) {
 	const router = useRouter()
+	const [grantReward, setGrantReward] = useState<number>(0)
+
+	const currentDate = new Date().getTime()
+
+	useEffect(() => {
+		if(grantReward === 0) {
+			calculateUSDValue(grantAmount, grantCurrency).then((promise: any) => {
```
Comment (adiwajshing) [inline src/components/profile/grantCard/index.tsx:64]: can just write `calculateUSDValue(grantAmount, grantCurrency).then(setGrantReward)`

File: src/components/signup/create_dao/form.tsx [hunk 9]
```diff
@@ -97,13 +131,14 @@ function Form({
...
 		}
 
-		onFormSubmit({
-			name: daoName,
-			description: daoDescription,
-			image: imageFile,
-			network: chainId!,
-			ownerId: accountData?.address ?? '0x0000000000000000000000000000000000000000',
-		})
+		// onFormSubmit({
```
Comment (adiwajshing) [inline src/components/signup/create_dao/form.tsx:134]: accidental?

File: src/utils/calculatingUtils.tsx [hunk 10]
```diff
@@ -0,0 +1,242 @@
+import { createClient } from 'urql'
+
+export const calculateUSDValue = async(value: number | string | any, token: string) => {
```
Comment (adiwajshing) [inline src/utils/calculatingUtils.tsx:3]: this function has a lot of duplicated code, and it won't work with new currencies added -- would recommend abstraction & modification to the `chains` repo -- so users can specify how to query the currency's USD value

**Pull request merged**